src/data/resolvers/motions.ts

```diff
@@ -0,0 +1,110 @@
+
+import {
+  ClientType,
+  getLogs,
+  getBlockTime,
+  ExtensionClient,
+} from '@colony/colony-js';
+
+import { Context } from '~context/index';
+import { ColonyAndExtensionsEvents } from '~types/index';
+import {
+  ActionsPageFeedType,
+  SystemMessage,
+  SystemMessagesName,
+} from '~dashboard/ActionsPageFeed';
+import { NetworkMotionState } from '~utils/colonyMotions';
+
+import { ProcessedEvent } from './colonyActions';
+
+export const motionsResolvers = ({
+  colonyManager: { networkClient },
+  colonyManager,
+}: Required<Context>): Resolvers => ({
+  Query: {
+    async motionsSystemMessages(_, { motionId, colonyAddress }) {
+      const { provider } = networkClient;
+      const votingReputationClient = (await colonyManager.getClient(
+        ClientType.VotingReputationClient,
+        colonyAddress,
+      )) as ExtensionClient;
+      const motion = await votingReputationClient.getMotion(motionId);
+      const motionNetworkState = await votingReputationClient.getMotionState(
+        motionId,
+      );
+      const systemMessages: SystemMessage[] = [];
+
+      // @ts-ignore
+      const motionStakedFilter = votingReputationClient.filters.MotionStaked(
+        motionId,
+        null,
+        null,
+        null,
+      );
+      const motionStakedLogs = await getLogs(
+        votingReputationClient,
+        motionStakedFilter,
+      );
+      // @ts-ignore
+      // eslint-disable-next-line max-len
+      const motionVoteRevealedFilter = votingReputationClient.filters.MotionVoteRevealed(
+        motionId,
+        null,
+        null,
+      );
+      const motionVoteRevealedLogs = await getLogs(
+        votingReputationClient,
+        motionVoteRevealedFilter,
+      );
+
+      const parsedEvents = await Promise.all(
+        [...motionStakedLogs, ...motionVoteRevealedLogs].map(async (log) => {
+          const parsedLog = votingReputationClient.interface.parseLog(log);
+          const { address, blockHash, blockNumber, transactionHash } = log;
+          const { name, values } = parsedLog;
+          return {
+            type: ActionsPageFeedType.NetworkEvent,
+            name,
+            values,
+            createdAt: blockHash ? await getBlockTime(provider, blockHash) : 0,
+            emmitedBy: ClientType.ColonyClient,
+            address,
+            blockNumber,
+            transactionHash,
+          } as ProcessedEvent;
+        }),
+      );
+
+      const sortedEvents = parsedEvents.sort(
+        (firstEvent, secondEvent) =>
+          secondEvent.createdAt - firstEvent.createdAt,
+      );
+
+      if (
+        motionNetworkState === NetworkMotionState.Finalizable ||
+        motionNetworkState === NetworkMotionState.Finalized
+      ) {
+        const newestStakeOrVoteEvent = sortedEvents.find(
+          (event) =>
+            event.name === ColonyAndExtensionsEvents.MotionStaked ||
+            event.name === ColonyAndExtensionsEvents.MotionVoteRevealed,
+        );
```

Comment:
Why are you looking for the `MotionStaked` / `MotionReveled` events in here ?

Wouldn't this work without checking for them?

I mean the motion is already in the `Finalizable`/`Finalized` state, meaning either we staked or voted, but either way we got to the end of the flow.

So we just need to decide if it passed or failed _(which you do, for both stakes or votes)_

So I think this will work even without searching for those two events.

Reply:
because i need timestamp of the newest event

Reply:
You are right. I haven't thought of that. 

Very nice! :+1: